ProcessLib/BoundaryCondition/NonuniformNeumannBoundaryCondition.cpp

```diff
@@ -8,33 +8,76 @@
  */
 
 #include "NonuniformNeumannBoundaryCondition.h"
+
+#include "MeshLib/IO/readMeshFromFile.h"
 #include "ProcessLib/Utils/ProcessUtils.h"
 
 namespace ProcessLib
 {
 std::unique_ptr<NonuniformNeumannBoundaryCondition>
 createNonuniformNeumannBoundaryCondition(
     BaseLib::ConfigTree const& config,
-    std::vector<MeshLib::Element*>&& elements,
     NumLib::LocalToGlobalIndexMap const& dof_table, int const variable_id,
-    int const component_id, bool is_axially_symmetric,
-    unsigned const integration_order, unsigned const shapefunction_order,
-    unsigned const global_dim,
-    std::vector<std::unique_ptr<ParameterBase>> const& parameters)
+    int const component_id, unsigned const integration_order,
+    unsigned const shapefunction_order, MeshLib::Mesh const& bulk_mesh)
 {
     DBUG("Constructing NonuniformNeumann BC from config.");
     //! \ogs_file_param{prj__process_variables__process_variable__boundary_conditions__boundary_condition__type}
     config.checkConfigParameter("type", "NonuniformNeumann");
 
-    //! \ogs_file_param{prj__process_variables__process_variable__boundary_conditions__boundary_condition__NonuniformNeumann__parameter}
-    auto const param_name = config.getConfigParameter<std::string>("parameter");
-    DBUG("Using parameter %s", param_name.c_str());
+    // TODO handle paths correctly
+    auto const mesh_file = config.getConfigParameter<std::string>("mesh");
+
+    std::unique_ptr<MeshLib::Mesh> surface_mesh(
+        MeshLib::IO::readMeshFromFile(mesh_file));
+
+    // Surface mesh and bulk mesh must have equal axial symmetry flags!
+    surface_mesh->setAxiallySymmetric(bulk_mesh.isAxiallySymmetric());
+
+    // TODO add field type?
+    auto const field_name =
+        config.getConfigParameter<std::string>("field_name");
+
+    auto const* const property =
+        surface_mesh->getProperties().getPropertyVector<double>(field_name);
+
+    if (!property)
+    {
+        OGS_FATAL("A property with name `%s' does not exist in `%s'.",
+                  field_name.c_str(), mesh_file.c_str());
+    }
+
+    if (property->getMeshItemType() != MeshLib::MeshItemType::Node)
+    {
+        OGS_FATAL(
+            "Only nodal fields are supported fur non-uniform fields. Field "
+            "`%s' is not nodal.",
+            field_name.c_str());
+    }
+
+    if (property->getNumberOfComponents() != 1)
+    {
+        OGS_FATAL("`%s' is not a one-component field.", field_name.c_str());
+    }
+
+    auto const* const mapping_to_bulk_nodes =
+        surface_mesh->getProperties().getPropertyVector<long>(
```

Comment:
I prepared the test data using a small paraview script: [click me](https://github.com/ufz/ogs-utils/blob/master/pre/BCs/inhomogeneous-mass-flux/prepare-inhomogeneous-mass-flux-BC.py).
Since VTK does not have a `std::size_t` data array (cf. [here](http://www.vtk.org/doc/nightly/html/classvtkDataArray.html)), I decided to use `unsigned long` in the script (vtkIdType didn't work for me). Apparently VTK does some casting, so I can read the `unsigned long` array as `long`.
So I can probably change the type to std::size_t.